test/test_segment_reductions.py

```diff
@@ -0,0 +1,33 @@
+import torch
+from torch.testing._internal.common_device_type import (
+    instantiate_device_type_tests,
+    onlyCPU,
+    dtypes,
+)
+from torch.testing._internal.common_utils import (
+    TestCase,
+    run_tests,
+)
+
+
+class TestSegmentReductions(TestCase):
+    @onlyCPU
+    @dtypes(torch.half, torch.bfloat16, torch.float, torch.double)
+    def test_max_simple_1d(self, device, dtype):
+        lengths = torch.tensor([1, 2, 3], device=device)
+        data = torch.tensor([1, float("nan"), 3, 4, 5, 6], device=device, dtype=dtype)
+        expected_result = torch.tensor([1, 3, 6], device=device, dtype=dtype)
```

Comment:
Typically, operations in pytorch propagate `nans` - do you need different behavior here?

Reply:
replied in earlier comment. Will update the PR to return nan instead of reduction based on remaining elements.